File: java/replicator/src/test/java/util/TestUtil.java
package util;

import command.Command;
import command.Command.CommandType;
import java.util.ArrayList;
import java.util.List;
import log.Instance;
import log.Instance.InstanceState;
import multipaxos.Configuration;

public class TestUtil {


  public static Instance makeInstance(long ballot, long index) {
    return new Instance(ballot, index, 0, InstanceState.kInProgress, new Command());
  }

  public static Instance makeInstance(long ballot, long index, CommandType type) {
    return new Instance(ballot, index, 0, InstanceState.kInProgress, new Command(type, "", ""));
  }

  public static Instance makeInstance(long ballot, long index, InstanceState state,
      CommandType type) {
    return new Instance(ballot, index, 0, state, new Command(type, "", ""));
  }

  public static Instance makeInstance(long ballot, long index, InstanceState state) {
    return new Instance(ballot, index, 0, state, new Command());
  }

  public static Configuration makeConfig(long id, int numPeers) {
    assert (id < numPeers);
    Configuration config = new Configuration();
    config.setId(id);
    config.setCommitInterval(300);
    config.setThreadPoolSize(8);
    List<String> peers = new ArrayList<>();
    for (int i = 0; i < numPeers; i++) {
      peers.add("127.0.0.1:1" + i + "000");
    }
    config.setPeers(peers);
    return config;
  }
}

File: c++-tcp/test_util.cc
#include <glog/logging.h>

#include "test_util.h"

Instance MakeInstance(int64_t ballot, int64_t index) {
  Instance i(ballot, index, 0, INPROGRESS, Command());
  return i;
}

Instance MakeInstance(int64_t ballot, int64_t index, InstanceState state) {
  auto i = MakeInstance(ballot, index);
  i.state_ = state;
  return i;
}

Instance MakeInstance(int64_t ballot, int64_t index, CommandType type) {
  auto i = MakeInstance(ballot, index);
  i.command_.type_ = type;
  return i;
}

Instance MakeInstance(int64_t ballot,
                      int64_t index,
                      InstanceState state,
                      CommandType type) {
  auto i = MakeInstance(ballot, index);
  i.state_ = state;
  i.command_.type_ = type;
  return i;
}

std::string MakeConfig(int64_t id, int64_t num_peers) {
  CHECK(id < num_peers);
  auto r = R"({ "id": )" + std::to_string(id) + R"(,
              "threadpool_size": 8,
              "commit_interval": 300,
              "store": "mem",
              "peers": [)";

  for (auto i = 0; i < num_peers; ++i) {
    r += R"("127.0.0.1:1)" + std::to_string(i) + R"(000")";
    if (i + 1 < num_peers)
      r += R"(,)";
  }
  r += R"(]})";

  return r;
}
